### ISSUE TITLE
I made some improvements

### ISSUE THREAD

**Issue by Baptiste-A** [danielkrupinski/OneByteRadar#4]
Hello,

I really like your cheat so I made improvements (only for usability) :

- some exception catching
- ability to enable and disable the cheat while in game (press p)

`
import pymem
import re
from pynput import keyboard
from termcolor import colored

running = False

try:  # checks if csgo is opened
    pm = pymem.Pymem('csgo.exe')

    def on_press(key):  # starts key listener
        try:
            k = key.char  # single-char keys
        except:
            k = key.name  # other keys
        if k in ['p']:  # keys of interest
            global running
            if not running:  # if cheat wasn't running, runs it
                print(colored('Running...', 'green'))
                running = True
            elif running:  # if cheat was running, stops it
                print(colored('Not running...', 'red'))
                running = False

            # --------------------------- cheat himself -------------------------------------
            pm = pymem.Pymem('csgo.exe')

            client = pymem.process.module_from_name(pm.process_handle,
                                                    'client.dll')

            clientModule = pm.read_bytes(client.lpBaseOfDll, client.SizeOfImage)
            address = client.lpBaseOfDll + re.search(rb'\x80\xB9.{5}\x74\x12\x8B\x41\x08',
                                                     clientModule).start() + 6

            pm.write_uchar(address, 0 if pm.read_uchar(address) != 0 else 2)

            pm.close_process()
            # ---------------------------- end of cheat -------------------------------------

    listener = keyboard.Listener(on_press=on_press)
    listener.start()  # start to listen on a separate thread
    listener.join()  # remove if main thread is polling self.keys

except pymem.exception.ProcessNotFound:  # if csgo isn't running,
                                         # catches the error and prints something readable
    print(colored("CS:GO isn't running...", 'red'))
`

**Comment by zlataovce** [comment]
Your code looks really great! Maybe you could make a pull request. I believe you have a mistake at the bottom except part, you have an indented comment right under the except: `# catches the error and prints something readable` . But if it works, it's good enough for me.

**Comment by Baptiste-A** [comment]
> 
> 
> Your code looks really great! Maybe you could make a pull request. I believe you have a mistake at the bottom except part, you have an over-indented comment right under the except: `# catches the error and prints something readable` . But if it works, it's good enough for me.

thanks man, I indented it too much on purpose in order for it to align to the one above it since it's the following. The code works very well, I use it on my smurf account almost every day and no VAC ban yet

**Comment by zlataovce** [comment]
From my experience, VAC tends to be very delayed with bans. It took my account 3 months to get banned. 

Maybe we could collaborate on a project :)
Keep up the great work!

**Comment by Baptiste-A** [comment]
> 
> 
> From my experience, VAC tends to be very delayed with bans. It took my account 3 months to get banned.
> 
> Maybe we could collaborate on a project :)
> Keep up the great work!

We will see... I will keep you guys up to date with my VAC situation.
Yes of course, I'll appreciate it :)

**Comment by vivxk** [comment]
hi does this still work or it's outdated??

**Comment by Baptiste-A** [comment]
> hi does this still work or it's outdated now??

Hello, yep still working like a charm :)

**Comment by vivxk** [comment]
ok thx mate :)

**Comment by joshgallantt** [comment]
how did you find \x80\xB9.{5}\x74\x12\x8B\x41\x08 ?

**Comment by Baptiste-A** [comment]
hello, I didn't found it, it was already in danielkrupinski's code

**Comment by markito12** [comment]
> hello, I didn't found it, it was already in danielkrupinski's code

You didnt get banned yet?

**Comment by Baptiste-A** [comment]
> > hello, I didn't found it, it was already in danielkrupinski's code
> 
> You didnt get banned yet?

Not yet :)

**Comment by slhrlx** [comment]
you get banned?

**Comment by Baptiste-A** [comment]
> you get banned?

Nope

**Comment by slhrlx** [comment]
I'm making my own cheat in python, and learning more about this language, thanks for you and all community what made projects like this as public!

**Comment by Baptiste-A** [comment]
> I'm making my own cheat in python, and learning more about this language, thanks for you and all community what made projects like this as public!

Np dude but I didn’t made most of the job

**Comment by Sniferrrr** [comment]
hi, how can i contact you? I need your help

**Comment by Sniferrrr** [comment]
Baptiste-A, hi, how can i contact you? I need your help

**Comment by Baptiste-A** [comment]
> Baptiste-A, hi, how can i contact you? I need your help

Hello, I don’t think I’ll be able to help you much because I didn’t code the cheat itself

**Comment by talk2g** [comment]
> > you get banned?
> 
> Nope

does this inject into cs like any other normal cheat? (example osiris)
or is it different since its on python?

**Comment by Baptiste-A** [comment]
> > > you get banned?
> > 
> > 
> > Nope
> 
> does this inject into cs like any other normal cheat? (example osiris)
> or is it different since its on python?

No, it doesn’t inject anything, it just reads the game data stored in the ram

**Comment by talk2g** [comment]
@Baptiste-A 
that means vac cant detect it?
since nothing is being modified at all?
or m i wrong

**Comment by Baptiste-A** [comment]
> @Baptiste-A
> that means vac cant detect it?
> since nothing is being modified at all?
> or m i wrong

The only thing that is being modified is the red dots on your radar but it’s client side so I dont think il will be detected soon

**Comment by talk2g** [comment]
thx for help

**Comment by GabrielDourado37** [comment]
@Baptiste-A were you banned for this?

**Comment by Baptiste-A** [comment]
No, but i'll not try it on cs2 if I were you ...

**Comment by vivxk** [comment]
has anyone tried it on cs2?
